models/order.js

```diff
@@ -12,7 +13,68 @@ const Entry = new Schema({
 const orderSchema = new Schema({
   id: String,
   status: { type: String, default: constants.OPEN },
-  items: [Entry]
+  entries: [Entry]
 });
 
+orderSchema.statics.getOrders = function() {
+  return getOrders.apply(this)
+  .then(getEntries)
+  .catch((e) => {
+    throw new Error(e);
+  });
+}
+
+function getOrders() {
+  const Order = this;
+
+  return new Promise((resolve, reject) => {
+    Order.find().exec((err, orders) => {
+      if(err) {
+        reject(err);
+      }
+
+      resolve(orders.map(toOrder));
+    });
+  });
+}
+
+function populateEntries(order) {
+  return new Promise((resolve, reject) => {
+    Item.populate(order.entries, [{ path: 'item_id', model: 'Item', select: 'name price -_id' }], (err, res) => {
```

Comment:
Yeah, def, whatever works. Probably not a big deal / stupid optimization, but I just meant to refactor a bit so we don't have to iterate twice. Something like:
```
function getEntries(orders) {
  return Promise.all(orders.map(populateEntries));
}

function populateEntries(order) {
  return new Promise((resolve, reject) => {
    Item.populate(order.entries, [{ ................ }], (err, res) => {
      if (err) reject(err);
      order.entries = res.map(toEntry);
      resolve(order);
    });
  });
}
```